lib/posix/pthread.c

```diff
@@ -848,4 +848,9 @@ static int posix_thread_pool_init(void)
 	return 0;
 }
 
+int pthread_equal(pthread_t pt1, pthread_t pt2)
+{
+	return (pt1 == pt2);
```

Comment:
nit: no brackets needed here, but I'm not going to stop this from going in now. If there is another review that requires another spin, please fix this too.